cuda/test/matrix/csr_kernels.cpp

```diff
@@ -429,23 +429,6 @@ TEST_F(Csr, SimpleApplyToCsrMatrixIsEquivalentToRef)
 }
 
 
-TEST_F(Csr, SimpleApplyToCsrMatrixIsEquivalentToRefUnsorted)
-{
-    set_up_apply_data(std::make_shared<Mtx::automatical>());
-    auto trans = mtx->transpose();
-    auto d_trans = dmtx->transpose();
-    gko::test::unsort_matrix(static_cast<Mtx *>(dmtx.get()), rand_engine);
-    gko::test::unsort_matrix(static_cast<Mtx *>(d_trans.get()), rand_engine);
-
-    mtx->apply(trans.get(), square_mtx.get());
-    dmtx->apply(d_trans.get(), square_dmtx.get());
-
-    GKO_ASSERT_MTX_NEAR(square_dmtx, square_mtx, 1e-14);
-    GKO_ASSERT_MTX_EQ_SPARSITY(square_dmtx, square_mtx);
-    ASSERT_TRUE(square_dmtx->is_sorted_by_column_index());
-}
-
-
```

Comment:
in cusparse, they denote the csr sorted by row and then by column
https://docs.nvidia.com/cuda/cusparse/index.html#csr-format
they may implement the spgemm with hash table, so that works in cuda.
but the current impl in dpcpp does not support unsorted

Reply:
To quote from the Csr documentation:
```
 * Both the SpGEMM and SpGEAM operation require the input matrices to be sorted
 * by column index, otherwise the algorithms will produce incorrect results.
 ```